port/unix/omrsysinfo.c

```diff
@@ -4465,70 +4446,104 @@ omrsysinfo_cgroup_subsystem_iterator_hasNext(struct OMRPortLibrary *portLibrary,
 		break;
-	case OMR_CGROUP_SUBSYSTEM_CPU :
+	case OMR_CGROUP_SUBSYSTEM_CPU:
 		subsystemMetricMap = omrCgroupCpuMetricMap;
 		break;
-	case OMR_CGROUP_SUBSYSTEM_CPUSET :
+	case OMR_CGROUP_SUBSYSTEM_CPUSET:
 		subsystemMetricMap = omrCgroupCpusetMetricMap;
 		break;
 	default:
 		rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+		state->count += 1;
 		goto _end;
 	}
 	subsystemMetricMapElement = &subsystemMetricMap[state->count];
 	currentElement = subsystemMetricMapElement->metricInfoElementList;
 	if (NULL == state->fileContent) {
-		if ((SINGLE_CGROUP_METRIC == subsystemMetricMapElement->metricElementsCount) && (NULL != currentElement->metricKeyInFile)) {
-			isSingleLineKeyValue = TRUE;
-		}
-		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, metricElement->value, &(*state).fileContent, isSingleLineKeyValue, sizeRef);
-		if (0 == rc) {
-			if(NULL == state->fileContent) {
-				state->multiLineCounter = SINGLE_CGROUP_METRIC;
-			} else {
-				state->multiLineCounter = 0;
-			}
-		} else {
+		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, &(state->fileContent), metricElement->value, sizeRef);
+
+		/* In error condition or single metric condition we increment the counter to continue to next metric */
+		if ((0 != rc) || (NULL == state->fileContent)) {
+			state->count += 1;
 			goto _end;
 		}
-		
+
+		state->multiLineCounter = 0;
 	}
+
 	if (state->multiLineCounter < subsystemMetricMapElement->metricElementsCount) {
-		char *tempBuff = portLibrary->mem_allocate_memory(portLibrary, 1024, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+		char *tempBuff = portLibrary->mem_allocate_memory(portLibrary, CGROUP_METRIC_FILE_CONTENT_MAX_LIMIT, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
 		char *part = "";
 		char *saveptr = NULL;
 		currentElement += state->multiLineCounter;
 		strcpy(tempBuff, state->fileContent);
 		part = strtok_r(tempBuff, "\n", &saveptr);
 	  	do {
-			if(0 == strncmp(part, currentElement->metricKeyInFile, strlen(currentElement->metricKeyInFile))) {
+			if (0 == strncmp(part, currentElement->metricKeyInFile, strlen(currentElement->metricKeyInFile))) {
 				part += strlen(currentElement->metricKeyInFile);
 				strcpy(metricElement->value, ++part);
-				rc = 0;
 				portLibrary->mem_free_memory(portLibrary, tempBuff);
-				state->multiLineCounter = state->multiLineCounter + 1;
-				goto _end;
+				rc = 0;
+				break;
 			}
 	  	} while ((part = strtok_r(NULL, "\n", &saveptr)) != NULL);
-	} else {
-		portLibrary->mem_free_memory(portLibrary, state->fileContent);
-		state->count = state->count + 1;
+
+		/* if we complete parse the fileContent and couldn't find the metric return error code */
+		if (NULL == part) {
+			rc = OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_METRIC_NOT_AVAILABLE;
+		}
+
+		state->multiLineCounter += 1;
```

Comment:
Trying to understand the `multi` part of this name.... Is it a lineCounter?  What makes it `multi` line?  Is there something that distinguishes single vs multi lines?